pkg/system/phase2_creating.go

```diff
@@ -433,6 +434,16 @@ func (r *Reconciler) setDesiredCoreEnv(c *corev1.Container) {
 			} else {
 				c.Env[j].Value = ""
 			}
+		case "CONFIG_JS_RESTRICT_RESOURCE_DELETION":
+			// check if provider mode is enabled and signal the core
+			annotationValue, annotationExists := util.GetAnnotationValue(r.NooBaa.Annotations, "MulticloudObjectGatewayProviderMode")
+			annotationBoolVal := false
+			if annotationExists {
+				annotationBoolVal = strings.ToLower(annotationValue) == strconv.FormatBool(true)
```

Comment:
The `strconv` is not so readable. Please use a const instead. You can define `trueStr` or a similar const in the file scope. Use it also for setting the env value below